pkgs/by-name/ff/fflogs/package.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, appimageTools
+, fetchurl
+}:
+
+let
+  pname = "fflogs";
+  version = "8.3.42";
+  src = fetchurl {
+    url = "https://github.com/RPGLogs/Uploaders-fflogs/releases/download/v${version}/fflogs-v${version}.AppImage";
+    sha256 = "08mkixjpppijssmg63l542xhmxyr7xs9z1v9qfjda7cfqgah5kh5";
+  };
+  extracted = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications
+    cp -r ${extracted}/usr/share/icons $out/share/
+    cp ${extracted}/fflogs.desktop $out/share/applications/
+    sed -i s@^Exec=.\*@Exec=$out/bin/fflogs@g $out/share/applications/fflogs.desktop
```

please avoid nix store paths in desktop entries

```suggestion
    sed -i s@^Exec=.\*@Exec=fflogs@g $out/share/applications/fflogs.desktop
```

---

done, though I'm curious about the reasoning

---

some desktop environments like gnome will copy dekstop entries to `~/.config/autostart` to schedule them for automatic start on login. Store paths may go out of date and get garbage collected, causing the startup entries to stop working.